include/nlohmann/detail/meta/type_traits.hpp

```diff
@@ -684,5 +684,57 @@ inline constexpr bool value_in_range_of(T val)
     return value_in_range_of_impl1<OfType, T>::test(val);
 }
 
+template<bool Value>
+using bool_constant = std::integral_constant<bool, Value>;
+
+///////////////////////////////////////////////////////////////////////////////
+// is_c_string
+///////////////////////////////////////////////////////////////////////////////
+
+namespace impl
+{
+
+template<typename T>
+inline constexpr bool is_c_string()
+{
+    using TUnExt = typename std::remove_extent<T>::type;
+    using TUnCVExt = typename std::remove_cv<TUnExt>::type;
+    using TUnPtr = typename std::remove_pointer<T>::type;
+    using TUnCVPtr = typename std::remove_cv<TUnPtr>::type;
+    return
+        (std::is_array<T>::value && std::is_same<TUnCVExt, char>::value)
```

Comment:
This will match `char *string` but not `unsigned char *` or `signed char *`.   Is that an issue?  Do we normally accept those types?

Reply:
Not for strings. And this check is only used to decide when the return type should be `string_t` instead of the given `ValueType`.

Reply:
Assuming  `unsigned char *` is `get()`-able, that would still work. Maybe there is for `binary_t`? I don't know.

Reply:
Okay, sounds like this would work as-is then.